### ISSUE TITLE
Atmosfera breaks sprint-swimming on NeoForge (via Sinytra Connector)

### ISSUE THREAD

**Issue by unilock** [Haven-King/Atmosfera#41]
With Atmosfera installed on NeoForge via [Sinytra Connector](https://modrinth.com/mod/connector), sprint-swimming becomes impossible - the player never enters that state, and continues swimming normally, regardless of whether the sprint key is held.

I suspect this mixin does not play well with NeoForge:

https://github.com/Haven-King/Atmosfera/blob/c358dae628994e566322051e726e4273b9ce07a7/src/main/java/dev/hephaestus/atmosfera/mixin/EntityMixin.java#L32-L56

I think this is probably an issue with Atmosfera instead of Sinytra Connector or NeoForge, as using `@Overwrite` is known to cause issues even with other Fabric mods.

***

For reference, this is what that method looks like on NeoForge:

```java
public boolean isEyeInFluid(TagKey<Fluid> fluidTag) {
	if (fluidTag == FluidTags.WATER) {
		return this.isEyeInFluidType(NeoForgeMod.WATER_TYPE.value());
	}
	if (fluidTag == FluidTags.LAVA) {
		return this.isEyeInFluidType(NeoForgeMod.LAVA_TYPE.value());
	}
	return this.fluidOnEyes.contains(fluidTag);
}
```

Since it looks like NeoForge already does what Atmosfera is trying to accomplish, perhaps it's enough to just disable Atmosfera's `EntityMixin` if it's detected to be running alongside Sintyra Connector?

**Comment by Fourmisain** [comment]
Oh yeah, that definitely shouldn't be an `@Overwrite`.
Can't say I fully understand what exactly it's trying to fix yet and if it's still needed.
(This is really old code, so it might have been fixed in Fabric API.)

> Since it looks like NeoForge already does what Atmosfera is trying to accomplish, perhaps it's enough to just disable Atmosfera's EntityMixin if it's detected to be running alongside Sintyra Connector?

I don't think NeoForge's code solves the issue as it's still only testing for identity.

**Comment by Fourmisain** [comment]
Yeah, this definitely isn't needed in newer versions. `TagRegistry` was deprecated in 1.18 and replaced by `TagFactory`, which was removed later (still trying to find when) and replaced by the vnailla `TagKey.of()`, which uses an interner, so checking for identity will always work.

**Comment by Fourmisain** [comment]
1.19 doesn't have `TagRegistry` or `TagFactory`.

Interestingly, even 1.18.2 has `TagKey.of()` (and I doubt Fabric API doesn't use that internally), so this might have only been an issue in 1.17 and below.

I'm just gonna remove this piece of code from 1.18+ versions.

**Comment by unilock** [comment]
> I don't think NeoForge's code solves the issue as it's still only testing for identity.

Oops, you're right, I missed where Atmosfera was checking against `TagKey#id`.

> Should be fixed in 2.0.1!

Awesome, thanks!
